pandas/core/indexes/base.py

```diff
@@ -1890,8 +1865,10 @@ def _concat_same_dtype(self, to_concat, name):
         # must be overridden in specific classes
         return _concat._concat_index_asobject(to_concat, name)
 
-    _index_shared_docs['take'] = """
-        return a new %(klass)s of the values selected by the indices
+    def take(self, indices, axis=0, allow_fill=True,
+             fill_value=None, **kwargs):
+        """
+        return a new Index of the values selected by the indices
```

Comment:
Thoughts on phrasing this as "Return a new index of the same type ..."? I suppose "of the values" implies that you're getting the same type of index back though.

Reply:
generally we need language to assert that we are getting back the same type of Index after the op (which is almost always true), though there are exceptions. This is technically easier to have the doc-strings render like this, e.g. ``PeriodIndex`` is pretty explicty as a return type for example.